### ISSUE TITLE
Special printing of DomainError broken by #14815

### ISSUE THREAD

**Issue by tkelman** [JuliaLang/julia#14985]
and not tested carefully enough, apparently
```
  | | |_| | | | (_| |  |  Version 0.5.0-dev+2532 (2016-02-08 03:26 UTC)
 _/ |\__'_|_|_|\__'_|  |  Commit 019c4d1* (0 days old master)
|__/                   |  x86_64-linux-gnu

julia> sqrt(-1)
ERROR: DomainError:SYSTEM: show(lasterr) caused an error
```
since `lookup` is not defined here https://github.com/JuliaLang/julia/blob/019c4d12861fc91fc73fca54123adb7f0a509a1c/base/replutil.jl#L127

**Comment by vtjnash** [comment]
fixed (and tested) in 69f75712916135e1f960246aa0b90169636e2a7f

**Comment by tkelman** [comment]
Great! Those don't all pass on Windows or Linux though.
